src/Quartz/Util/StringKeyDirtyFlagMap.cs

```diff
@@ -347,5 +356,39 @@ public virtual TimeSpan GetTimeSpan(string key)
             object obj = this[key];
             return (TimeSpan) obj;
         }
+
+        /// <summary>
+        /// Retrieve the identified <see cref="Guid" /> value from the <see cref="JobDataMap" />.
+        /// </summary>
+        public virtual Guid GetGuid(string key)
+        {
+            object obj = this[key];
+
+            try
+            {
+                return (Guid)obj;
+            }
+            catch (Exception)
```

Comment:
This try..catch adds 7 lines of code just to change "System.InvalidCastException: Specified cast is not valid." into "System.InvalidCastException: Identified object is not a Guid.".

That seems a bit overkill.

Reply:
Thanks for pointing this out, I have removed the try/catch blocks so it throws the original exception.